src/components/transport_manager/src/transport_manager_impl.cc

```diff
@@ -673,21 +712,204 @@ TransportManagerImpl::ConnectionInternal* TransportManagerImpl::GetConnection(
+      : what_(what) {}
+  bool operator()(const SwitchableDevices::value_type& i) const {
+    return what_->second == i.second;
+  }
+
+ private:
+  SwitchableDevices::const_iterator what_;
+};
+
+}  // namespace
+
+void TransportManagerImpl::TryDeviceSwitch(
+    transport_adapter::TransportAdapter* adapter) {
+  LOG4CXX_AUTO_TRACE(logger_);
+  if (adapter->GetDeviceType() != transport_adapter::DeviceType::IOS_USB) {
+    LOG4CXX_ERROR(logger_, "Switching requested not from iAP-USB transport.");
+    return;
+  }
+
+  const auto ios_bt_adapter = std::find_if(transport_adapters_.begin(),
+                                           transport_adapters_.end(),
+                                           IOSBTAdapterFinder());
+
+  if (transport_adapters_.end() == ios_bt_adapter) {
+    LOG4CXX_WARN(
+        logger_,
+        "There is no iAP2 Bluetooth adapter found. Switching is not possible.");
+    return;
+  }
+
+  const SwitchableDevices usb_switchable_devices =
+      adapter->GetSwitchableDevices();
+  const auto bt_switchable_devices = (*ios_bt_adapter)->GetSwitchableDevices();
+  auto bt = bt_switchable_devices.end();
+  auto usb = usb_switchable_devices.begin();
+  for (; usb != usb_switchable_devices.end(); ++usb) {
+    SwitchableFinder finder(usb);
+    bt = std::find_if(
+        bt_switchable_devices.begin(), bt_switchable_devices.end(), finder);
+
+    if (bt != bt_switchable_devices.end()) {
+      break;
+    }
+  }
+
+  if (bt_switchable_devices.end() == bt) {
+    LOG4CXX_WARN(logger_,
+                 "No suitable for switching iAP2 Bluetooth device found.");
+    return;
+  }
+
+  LOG4CXX_DEBUG(logger_,
+                "Found UUID suitable for transport switching: " << bt->second);
+  LOG4CXX_DEBUG(
+      logger_, "Device to switch from: " << bt->first << " to: " << usb->first);
+
+  sync_primitives::AutoWriteLock lock(device_to_adapter_map_lock_);
+
+  const auto bt_device_uid = bt->first;
+  const auto device_to_switch = device_to_adapter_map_.find(bt_device_uid);
+  if (device_to_adapter_map_.end() == device_to_switch) {
+    LOG4CXX_ERROR(logger_,
+                  "There is no known device found with UID "
+                      << bt_device_uid
+                      << " . Transport switching is not possible.");
+    DCHECK_OR_RETURN_VOID(false);
+    return;
+  }
+
+  const auto usb_uid = usb->first;
+  const auto bt_uid = device_to_switch->first;
+  const auto bt_adapter = device_to_switch->second;
+
+  LOG4CXX_DEBUG(logger_,
+                "Known device with UID "
+                    << bt_uid << " is appropriate for transport switching.");
+
+  RaiseEvent(
+      &TransportManagerListener::OnDeviceSwitchingStart, bt_uid, usb_uid);
+
+  bt_adapter->StopDevice(bt_uid);
+  adapter->DeviceSwitched(usb_uid);
+
+  // TODO(AOleynik): Check if next line functionality is still needed
+  DeactivateDeviceConnections(bt_uid);
+
+  // TODO(AOleynik): Check if next line functionality is still needed
```

Comment:
Is this functionality still needed?

Reply:
@JackLivio Fixed in 1d710a202cfed11af514ad375019ca284afd9001